.github/workflows/ci.yml

```diff
@@ -0,0 +1,56 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main   
+  pull_request:  
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    env:
+        RUSTFLAGS: "-D warnings"
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: stable
+          components: clippy
+
+      - name: Install cargo-hack
+        uses: taiki-e/install-action@v2
+        with:
+          tool: cargo-hack
+          checksum: true
+
+      - name: Check compilation
+        run: cargo hack check --all-features --all-targets
+
+      - name: Run Clippy
+        run: cargo clippy --all-features --all-targets
+
+  formatting:
+    name: Check formatting
+    runs-on: ubuntu-latest
+    needs: build  # Ensures this job runs after the build job
```

Comment:
This is not necessary, the formatting job can run in parallel to the build job. We can remove this line.